site/_posts/2017-05-07-0.3-release.md

```diff
@@ -0,0 +1,253 @@
+contributed integration between Arrow buffers and the Python buffer protocol
+and memoryviews, so now code like this is possible:
+
+```python
+In [6]: import pyarrow as pa
+
+In [7]: buf = pa.frombuffer(b'foobarbaz')
+
+In [8]: buf
+Out[8]: <pyarrow._io.Buffer at 0x7f6c0a84b538>
+
+In [9]: memoryview(buf)
+Out[9]: <memory at 0x7f6c0a8c5e88>
+
+In [10]: buf.to_pybytes()
+Out[10]: b'foobarbaz'
+```
+
+We have significantly expanded [**Apache Parquet**][21] support via the C++
+Parquet implementation [parquet-cpp][23]. This includes support for partitioned
+datasets on disk or in HDFS. We added initial Arrow-powered Parquet support [in
+the Dask project][24], and look forward to more collaborations with the Dask
+developers on distributed process of pandas data.
+
+With Arrow's support for pandas maturing, we were able to merge in the
+[**Feather format**][22] implementation, which is essentially a special case of
+the Arrow random access format. We'll be continuing Feather development within
+the Arrow codebase. For example, Feather can now read and write with Python
+file objects using Arrow's Python binding layer.
+
+We also implemented more robust support for pandas-specific data types, like
+`DatetimeTZ` and `Categorical.
+
+### Support for Tensors and beyond in C++ Library
+
+There has been increased interest in using Apache Arrow as an tool for
+zero-copy shared memory management for machine learning applications. A
+flagship example is the [Ray project][26] from the UC Berkeley [RISELab][27].
+
+Machine learning deals in additional kinds of data structures beyond what the
+Arrow columnar format supports, like multidimensional arrays aka "tensors". As
+such, we implemented the [`arrow::Tensor`][28] C++ type which can utilize the
+rest of Arrow's zero-copy shared memory machinery (using `arrow::Buffer` for
+managing memory lifetime). In C++ in particular, we will want to provide for
+additional data structures utilizing common IO and memory management tools.
+
+### Start of JavaScript (TypeScript) Implementation
+
+[Brian Hulette][20] started developing an Arrow implementation in
+[TypeScript][21] for use in NodeJS and browser-side applications. We are
+benefitting from Flatbuffers' first class support for JavaScript.
+
+### Improved Website and Developer Documentation
+
+Since 0.2.0 we have implemented a new website stack for publishing
+documentation and blogs based on [Jekyll][18]. Kouhei Sutou developed a [Jekyll
+Jupyter Notebook plugin][19] so that we can use Jupyter to author content for
+the Arrow website.
+
+On the website, we have now published API documentation for the C, C++, Java,
+and Python subcomponents. Within these you will find easier-to-follow developer
+instructions for getting started.
+
+### Contributors
+
+Thanks to all who contributed patches to this release.
+
+```
+$ git log apache-arrow-0.2.0..apache-arrow-0.3.0 --pretty=format:%aN | sort | uniq -c | sort -rn
```

Comment:
Ah, it's actually `git shortlog -sn apache-arrow-0.2.0..apache-arrow-0.3.0`, the `-c` flag will count by committer instead of author.